.github/workflows/python-unit-testing.yml

```diff
@@ -0,0 +1,49 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Python Services Unit Tests
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Install dependencies (Keywords)
+      working-directory: ./services/keywords
+      run: |
+        python -m pip install --upgrade pip
+        pip install pytest pytest-mock
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+        
+    - name: Install dependencies (Search Engine)
+      working-directory: ./services/search/
+      run: |
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+        
+    - name: Install dependencies (Parser)
+      working-directory: ./services/parser-database
+      run: |
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+  
+    - name: Test with pytest (parser)
+      working-directory: ./services/parser-database
+      run: |
+        python -m pytest
+    - name: Test with pytest (keyword)
+      working-directory: ./services/keywords
+      run: |
+        python -m pytest
+    - name: Test with pytest (searchengine)
+      working-directory: ./services/search
+      run: |
+        python -m pytest
```

Comment:
This actually seems like a great use case for a [matrix strategy](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategy). That way you can define a generic job template that will get executed for each service.